vizro-ai/tests/integration/test_dashboard.py

```diff
@@ -51,3 +57,81 @@ def test_simple_dashboard(dash_duo):
     app = Vizro().build(dashboard).dash
     dash_duo.start_server(app)
     assert dash_duo.get_logs() == []
+
+    pages = [1 if dashboard.pages else 0][0]
+    pages_num = [1 if len(dashboard.pages) == 2 else 0][0]
+
+    try:
+        grid_1 = [1 if dashboard.pages[0].layout.grid[0] == [0] else 0][0]
+    except IndexError:
+        grid_1 = 0
+    try:
+        grid_2 = [1 if dashboard.pages[1].layout.grid[0] == [0, 1, 1] else 0][0]
+    except IndexError:
+        grid_2 = 0
+
+    try:
+        components_num_1 = [1 if len(dashboard.pages[0].components) == 1 else 0][0]
+    except IndexError:
+        components_num_1 = 0
+    try:
+        components_num_2 = [1 if len(dashboard.pages[1].components) == 3 else 0][0]
+    except IndexError:
+        components_num_2 = 0
+
+    try:
+        controls_num_1 = [1 if len(dashboard.pages[0].controls) == 0 else 0][0]
+    except IndexError:
+        controls_num_1 = 0
+    try:
+        controls_num_2 = [1 if len(dashboard.pages[1].controls) == 2 else 0][0]
+    except IndexError:
+        controls_num_2 = 0
+
+    try:
+        comps_1 = [components.type for components in dashboard.pages[0].components]
+        components_types_1 = [1 if comps_1.count("ag_grid") == 1 else 0][0]
+    except IndexError:
+        components_types_1 = 0
+    try:
+        comps_2 = [components.type for components in dashboard.pages[1].components]
+        components_types_2 = [1 if comps_2.count("card") == 2 and comps_2.count("graph") == 1 else 0][0]
+    except IndexError:
+        components_types_2 = 0
+
+    try:
+        cntrls_2 = [controls.type for controls in dashboard.pages[1].controls]
+        controls_types_2 = [1 if cntrls_2.count("filter") == 2 else 0][0]
+    except IndexError:
+        controls_types_2 = 0
+
+    prescore = [
+        pages,
+        pages_num,
+        grid_1,
+        grid_2,
+        components_num_1,
+        components_num_2,
+        controls_num_1,
+        controls_num_2,
+        components_types_1,
+        components_types_2,
+        controls_types_2,
+    ]
+    score = sum(prescore)
+
+    with open(f"report_model_{model_name}.csv", "a", newline="") as csvfile:
+        writer = csv.writer(csvfile, delimiter=",", quotechar="|", quoting=csv.QUOTE_MINIMAL)
```

Comment:
why do we need "," and "|" in the csv? In this case, the result when open up from excel is not readable.

Maybe only keep the ","? something like:
```
    with open(f"report_model_{model_name}.csv", "a", newline="") as csvfile:
        writer = csv.writer(csvfile)
        writer.writerow(["Description", "Score"])
        writer.writerow(["Pages exists", str(pages)])
        writer.writerow(["Correct pages number", str(pages_num)])
        writer.writerow(["Correct grid first page", str(grid_1)])
        writer.writerow(["Correct grid second page", str(grid_2)])
        writer.writerow(["Correct components number first page", str(components_num_1)])
        writer.writerow(["Correct components number second page", str(components_num_2)])
        writer.writerow(["Correct controls number first page", str(controls_num_1)])
        writer.writerow(["Correct controls number second page", str(controls_num_2)])
        writer.writerow(["Correct components types first page", str(components_types_1)])
        writer.writerow(["Correct components types second page", str(components_types_2)])
        writer.writerow(["Correct controls types second page", str(controls_types_2)])
        writer.writerow(["Total", f"'{score}/{len(prescore)}'"])
```
to get rid of "|"

Reply:
Done